lib/task/digitalobject/digitalObjectRegenDerivativesTask.class.php

```diff
@@ -174,9 +174,28 @@ protected function execute($arguments = array(), $options = array())
 
   public static function regenerateDerivatives(&$digitalObject, $type = null)
   {
+    // Determine usage ID for type
+    switch($type)
+    {
+      case "reference":
+        $usageId = QubitTerm::REFERENCE_ID;
+        break;
+
+      case "thumbnail":
+        $usageId = QubitTerm::THUMBNAIL_ID;
+        break;
+
+      default:
+        $usageId = QubitTerm::MASTER_ID;
+    }
+
     // Delete existing derivatives
     $criteria = new Criteria;
-    $criteria->add(QubitDigitalObject::PARENT_ID, $digitalObject->id);
+    $criteria->add(QubitDigitalObject::USAGE_ID, $usageId);
```

Comment:
No, this will delete *ALL* derivatives in the application of the given type when `$type == thumbnail` or `$type == reference`.  :fearful:  

When no $type is specified we want to regenerate the thumbnail and reference derivatives for _this $digitalObject only_  I would recommend:

    $criteria->add(QubitDigitalObject::PARENT_ID, $digitalObject->id);
    if (isset($usageId))
    {
      $criteria->add(QubitDigitalObject::USAGE_ID, $usageId);
    }

And delete the `default:` clause from your switch statement (lines 187-189) so `$usageId = NULL`